### PR TITLE
DT-2509 Add ability to update Role Description

### PR DESCRIPTION
**Pull request by karenmillermoj**

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/oauth2server/resource/api/RolesController.kt [hunk 1]
```diff
@@ -228,10 +253,18 @@ data class RoleDetails(
...
   @field:Size(min = 4, max = 100)
   @field:Pattern(regexp = "^[0-9A-Za-z- ,.()'&]*\$")
   val roleName: String,
 )
+
+@ApiModel(description = "Role Description")
+data class RoleDescriptionAmendment(
+  @ApiModelProperty(required = true, value = "Role Description", example = "Maintaining admin users")
+  @field:Size(max = 1024)
+  @field:Pattern(regexp = "^[0-9A-Za-z- ,.()'&]*\$")
```
Comment (petergphillips) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/oauth2server/resource/api/RolesController.kt:268]: were you allowing newlines?

**Review by petergphillips — APPROVED** [review]
LGTM

File: src/main/kotlin/uk/gov/justice/digital/hmpps/oauth2server/resource/api/RolesController.kt (hunk 1 above)
Comment (karenmillermoj) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/oauth2server/resource/api/RolesController.kt:268]: yes, next PR - good spot!

**Review by andymarke — APPROVED** [review]
LGTM

**Review by petergphillips — APPROVED** [review]
LGTM

**Pull request merged**